packages/components/src/internal/components/assay/AssayPicker.test.tsx

```diff
@@ -60,4 +75,29 @@ describe('AssayPicker', () => {
         expect(document.querySelectorAll('#assay-type-select-container')).toHaveLength(0);
         expect(document.querySelectorAll('.alert-info')).toHaveLength(1);
     });
+
+    test('AssayPicker General provider only', async () => {
+        renderWithAppContext(
+            <AssayPicker
+                defaultTab={AssayPickerTabs.SPECIALTY_ASSAY_TAB}
+                hasPremium={false}
+                loadOptions={loadGeneralOnly}
+                onChange={jest.fn()}
+                showContainerSelect={false}
+                showImport={false}
+            />
+        );
+
+        await waitFor(() => {
+            expect(document.querySelectorAll('.nav-tabs li')).toHaveLength(2);
+        });
+
+        // Verify only two tabs and specialty tab selected
```

Comment:
I see you're setting the default tab to the specialty tab above, but if there are no specialty assays, do we expect that tab to be selected in the app?

Reply:
No, the app won't select that tab by default in this scenario. The tab is clickable so the user would see a message about "This server does not have any specialty or custom assay types.". This test just uses that prop to trigger the state loading for the relevant code, instead of having to simulate a tab click.